### ISSUE TITLE
Wireguard does not start when using dns name instead ip

### ISSUE THREAD

**Issue by TomFreudenberg** [opnsense/plugins#3203]
This is a special condition, when using following configuration:

1. Unbound
2. Bind
3. WireGuard

If you run your domain via your BIND on the same OpnSense as Wireguard and use a name from that domain for your endpoint, the service start for wireguard will fail (mostly) on reboot.

The rc.d order of starting services says for named:

```
# PROVIDE: named
# REQUIRE: NETWORKING ldconfig syslogd
# BEFORE: SERVERS
```

and for unbound:

```
# PROVIDE: unbound
# REQUIRE: FILESYSTEMS defaultroute netwait resolv
# BEFORE: NETWORKING
```

and for wireguard / openvpn:

```
# PROVIDE: wireguard
# REQUIRE: NETWORKING
```

Here the wireguard service may try to start BEFORE named is running.

If it would be for named (changed BEFORE list):

```
# PROVIDE: named
# REQUIRE: NETWORKING ldconfig syslogd
# BEFORE: SERVERS wireguard
```

all will run fine.

Is this suitable for a PR or change or how to you think about it?

Cheers
Tom

**Comment by fichtner** [comment]
Hi Tom, this would be for FreeBSD ports to decide. But given the basic issue of DNS vs raw address changing the order for one use case will break other use cases, e.g. where VPN tunnels are required before DNS resolution because they is moved over the tunnel.

typically use use a late additional start script, a few plugins do this.


Cheers,
Franco

**Comment by TomFreudenberg** [comment]
Hi Franco @fichtner 

yes, same I thought - it's some kind of rare condition and depends on the specific system configuration.

Do you know if already a plugin exist to manage / adjust those orders for service startup?

**Comment by fichtner** [comment]
actually the workaround is in place: https://github.com/opnsense/plugins/commit/fed5c886cf3b4d

are you using go or kmod? do you have more than one wireguard instance and it's already partially up?

**Comment by fichtner** [comment]
sorry I misread... correct file, partial command only

it also needs the following adjusted for wireguard before restarting the routing

https://github.com/opnsense/plugins/blob/fed5c886cf3b4d71a3f8f00a0b376be4a219f59b/net/ntopng/src/etc/rc.syshook.d/start/50-ntopng#L3-L4

**Comment by TomFreudenberg** [comment]
> are you using go or kmod?

currently just the standard with go

> do you have more than one wireguard instance

just one for site 2 site tunnel

>  it's already partially up?

what do you mean ?

---

Should I once patch my local installation with:

a. https://github.com/opnsense/plugins/commit/fed5c886cf3b4d

and

b. https://github.com/opnsense/plugins/blob/fed5c886cf3b4d71a3f8f00a0b376be4a219f59b/net/ntopng/src/etc/rc.syshook.d/start/50-ntopng#L3-L4

I can do this if you want me to test?

**Comment by fichtner** [comment]
> > it's already partially up?

> what do you mean ?

related to two instances question only ;)

> I can do this if you want me to test?

Yes, just add the (additional) wireguard start to 50-wireguard file before routing configuration. It should do the trick.

**Comment by TomFreudenberg** [comment]
The script 50-wireguard **was** already on my system 22.7.8

The script 50-ntopng does not exist

**Comment by fichtner** [comment]
yes, 50-ntopng was an example of what the workaround looks like for a service ;)

**Comment by TomFreudenberg** [comment]
Ok, so the 50-wireguard script does not do the trick.

Having this in place won't let the wireguard service start on boot

Currently for me the patch in rc.d named does the trick only or changing the endpoint address into an ip

**Comment by fichtner** [comment]
Starting wireguard prevents starting wireguard? I'm a bit lost here..

**Comment by TomFreudenberg** [comment]
:-) mee too

Just start again:

The 50-wireguard script already exist on my installation 22.7.8

If I do not change (patch) anything from installation and using a dns name from a bind master zone for endpoint, the wireguard service does not start on reboot

If I patch the rc.d for named and change to `BEFORE: SERVERS wireguard` all runs fine as expected.

**Comment by fichtner** [comment]
That's because the existing start script 50-wireguard **does not start wireguard** (yet). It was installed a long time ago for a slightly related purpose but misses the service start. I pointed this out, and how a service start looks like from a ntopng perspective, which is the type of workaround that I meant. I can't post a workaround of wireguard that does not exist yet. :)

**Comment by TomFreudenberg** [comment]
So should I try with replacement of 50-wireguard like

```bash
/usr/local/etc/rc.d/wireguard restart
```

?

**Comment by TomFreudenberg** [comment]
test that - mom

**Comment by TomFreudenberg** [comment]
No, does not work.

Right after start a number of services are not available on this:

![image](https://user-images.githubusercontent.com/410087/204009870-b3cdb7ce-f254-43d6-8794-4952e007842b.png)

All that is not happen when changing the rc.d BEFORE or direct setting an ip address

**Comment by TomFreudenberg** [comment]
After 2 min 30 its all started

![image](https://user-images.githubusercontent.com/410087/204010121-a2e06d3d-9f40-4633-9fd1-0d8c17202d6f.png)

Hmm seems to do the job

**Comment by TomFreudenberg** [comment]
When I use the BEFORE patch or an ip al services are available right after boot.

But in case that it won't started - having the line in 50-wireguard does at least work

50-wireguard

```bash
#!/bin/sh

/usr/local/etc/rc.d/wireguard restart

# reconfigure routing for wireguard
/usr/local/etc/rc.routing_configure
```

**Comment by fichtner** [comment]
Well, we can issue a second start but we have no influence on individual boot times :)

**Comment by TomFreudenberg** [comment]
Understand - so you will decide how and if to fix?

For me than issue can be closed

Thanks for discussion and help

**Comment by fichtner** [comment]
Patched via 2cdaf71, thanks!
